### PR TITLE
[WIP] Crowdfunding tests

### PR DESCRIPTION
**Pull request by pozylon**
- [x] Copy test setup from publikator-backend
- [x] Clean up the test util code and move the general tear up/down helpers into an own file
- [x] Rerun tests with nodemon on file change to speed up development a bit
- [ ] Find out why either yarn or tape do not like the way I defined the package.json test?:* commands.
- [x] Improve seeds/fixtures and launch scripts regarding testing
- [ ] Tests: pledges & payments
- [ ] Tests: claim memberships
- [ ] Tests: user-scoped queries

### CODE REVIEW

**Comment by patte** [comment]
I split up the webhook handlers, so you are free to after them!

File: modules/crowdfundings/__tests__/payPledge.test.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+const PAY_PLEDGE_MUTATION = `
+  mutation payPledge($pledgeId: ID!, $method: PaymentMethod!, $sourceId: String, $pspPayload: String, $address: AddressInput, $paperInvoice: Boolean) {
+    payPledge(pledgePayment: {pledgeId: $pledgeId, method: $method, sourceId: $sourceId, pspPayload: $pspPayload, address: $address, paperInvoice: $paperInvoice}) {
+      pledgeId
+      userId
+      emailVerify
+    }
+  }
+`
```
Comment (patte) [inline modules/crowdfundings/__tests__/payPledge.test.js:32]: just fyi: it's save to split the arguments on multiple lines, even without `,`

**Comment by patte** [comment]
looking very nice so far

**Comment by patte** [comment]
Looking very nice so far! 👍 
Feedback / Todos / Comments:
- I think it would be better if you would work on a local branch. This way the base would update automatically, thus avoiding merge conflicts and others can easily make a branch based on your one.
- The seeds are awesome! It would be cool if you can add a test command that clears the DB and loads them. Have a look at [test](https://github.com/orbiting/publikator-backend/blob/master/package.json#L73) in the publikator-backend. Or just shoot at the DB from the [tests](https://github.com/orbiting/republik-backend/blob/6845b707b3784233c66d6b3a455df1833c22fbe1/__tests__/discussion/topLevel.test.js#L21)
- ~~Also in the publikator-backend you find all the necessities to make it run on travis. I invited you to our github org, so you should be able to change settings on travis if needed.~~ done in: https://github.com/orbiting/republik-backend/pull/27/commits/032d7fd06e5570277aa927cd6357d979ac36e38b
- You split up the tests into multiple files 👍 How do you make sure integration.test.js get's executed before other tests? Do you already have an idea how to shutdown the server after all the tests in all the files have run?  

Also, I created this [PR](https://github.com/orbiting/republik-backend/pull/27) and shamelessly stole your test setup (just moved things out of modules eg. `__tests__/helpers.js`). If you move to a local branch, I can base mine on yours or vice versa.

**Comment by pozylon** [comment]
Thanks! Yes I will definitely move to a local branch and will try to come up with the same set of operations/commands compared to the publikator-backend so we all know how to start, stop, init tests in most repos and the travis configs can stay consistent.

Regarding the multiple files. It doesn't matter which file comes first as the setup should make it possible to run every test in isolation as the global state is re-initiated at the beginning of every test (connectIfNeeded).

But it does matter which one comes last to stop the test :( i didn't try to use https://github.com/substack/tape#testonfinishfn yet though.

**Comment by pozylon** [comment]
Closing in favor of #28

**Pull request closed without merging**